### PR TITLE
docs: fix broken charts links

### LINKED ISSUE #3910: [Broken link] Link to network diagram page goes to 404
## Detailed description

In the V11 site, when I click the tile for Network diagram on the Data Visualizations/Chart types page, I get a 404 error.
This happens for all other the complex charts types as well. 

## Steps to reproduce the issue

1. Step one: Go to [Chart types](https://carbondesignsystem.com/data-visualization/chart-types/) page
3. Step two: Select [Network diagram](https://carbondesignsystem.com/data-visualization/complex-charts#network-diagrams)
4. Step three: Receive 404 page:
Screenshot 2024-02-12 at 12 24 51 PM

## Additional information

The [Complex charts / Network diagram](https://v10.carbondesignsystem.com/data-visualization/complex-charts#network-diagrams) page can still be accessed correctly in the v10 version of the Carbon site.

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes  #3910 

- Fixes a broken link mentioned in the issue specifically for the "Network diagrams"card on the Data visualizations > [Chart types page](https://carbondesignsystem.com/data-visualization/chart-types/)
- Fixes several other broken links for these cards on this page:
  - Tree map
  - Circle pak
  - Heat map
  - Parallel coordinates
  - Alluvial diagram
  - Tree diagram
  - Choropleth map
  - Proportional symbol
  - Connecting lines

------

**Changed**

- Fixed links that were 404ing

### CODE REVIEW

**Review by guidari — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

**Comment by Kritvi-bhatia17** [comment]
Links look perfect now!! 
Nice work @laurenmrice 

I have a small question. When we click on [Connecting Links](https://carbondesignsystem-git-laurenmrice-patch-3-carbon-design-system.vercel.app/data-visualization/spatial-charts/#connection-map), it redirects to [Connection Map](https://carbondesignsystem-git-laurenmrice-patch-3-carbon-design-system.vercel.app/data-visualization/spatial-charts/#connection-map). Shouldn't the heading of one of them be adjusted / updated for consistency? 🤔

**Review by Kritvi-bhatia17 — COMMENTED** [review]
Links look perfect now!!
Nice work @laurenmrice

I have a small question. When we click on [Connecting Links](https://carbondesignsystem-git-laurenmrice-patch-3-carbon-design-system.vercel.app/data-visualization/spatial-charts/#connection-map), it redirects to [Connection Map](https://carbondesignsystem-git-laurenmrice-patch-3-carbon-design-system.vercel.app/data-visualization/spatial-charts/#connection-map). Shouldn't the heading of one of them be adjusted / updated for consistency? 🤔

**Comment by aagonzales** [comment]
@Kritvi-bhatia17 where in the page are you seeing the "connecting links" text?

File: src/pages/data-visualization/chart-types/index.mdx [hunk 1]
```diff
@@ -352,7 +352,7 @@ appropriate chart type.
 <Column  colMax={3} colXl={3} colLg={4} colMd={4} colSm={2} noGutterSm >
  <OverviewCard
     title="Connecting lines"
```
Comment (aagonzales) [inline src/pages/data-visualization/chart-types/index.mdx:354]: Here is where Kritvi mentioned the misalignment 
```suggestion
    title="Connection map"
```

**Review by aagonzales — APPROVED** [review]
Looks good except for the one quick fix

**Comment by laurenmrice** [comment]
@Kritvi-bhatia17 Oh yes, good call! And thank you @aagonzales !

**Review by Kritvi-bhatia17 — APPROVED** [review]
Perfect!!

**Pull request merged**